backend/api/Services/MissionDefinitionService.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System.Linq.Dynamic.Core;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using Api.Controllers.Models;
+using Api.Database.Context;
+using Api.Database.Models;
+using Api.Utilities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Api.Services
+{
+    public interface IMissionDefinitionService
+    {
+        public abstract Task<MissionDefinition> Create(MissionDefinition mission);
+
+        public abstract Task<MissionDefinition?> ReadById(string id);
+
+        public abstract Task<PagedList<MissionDefinition>> ReadAll(MissionDefinitionQueryStringParameters parameters);
+
+        public abstract Task<MissionDefinition> Update(MissionDefinition mission);
+
+        public abstract Task<MissionDefinition?> Delete(string id);
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1309:Use ordinal StringComparison",
+        Justification = "EF Core refrains from translating string comparison overloads to SQL"
+    )]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1304:Specify CultureInfo",
+        Justification = "Entity framework does not support translating culture info to SQL calls"
+    )]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "Globalization",
+        "CA1307:Specify CultureInfo",
+        Justification = "Entity framework does not support translating culture info to SQL calls"
+    )]
+    public class MissionDefinitionService : IMissionDefinitionService
+    {
+        private readonly FlotillaDbContext _context;
+
+        public MissionDefinitionService(FlotillaDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<MissionDefinition> Create(MissionDefinition mission)
+        {
+            await _context.MissionDefinitions.AddAsync(mission);
+            await _context.SaveChangesAsync();
+
+            return mission;
+        }
+
+        private IQueryable<MissionDefinition> GetMissionsWithSubModels()
+        {
+            return _context.MissionDefinitions
+                .Include(mission => mission.Area)
+                .ThenInclude(robot => robot.Deck)
+                .ThenInclude(robot => robot.Installation)
+                .ThenInclude(robot => robot.Asset)
+                .Include(mission => mission.Source)
+                .Include(mission => mission.LastRun)
+                .ThenInclude(planTask => planTask == null ? null : planTask.StartTime);
```

Comment:
Not certain what this line is supposed to do in this setting? A `LastRun` is a `MissionRun` which does not have something called `PlanTask`